text/0000-delete-and-delete_by.md

```diff
@@ -0,0 +1,75 @@
+- Feature Name: Add `delete` and `delete_by` to `Iterator`
+- Start Date: 2018-06-14
+- RFC PR:
+- Rust Issue:
+
+# Summary
+[summary]: #summary
+This RFC asks to add two methods to `std::iter::Iterator`, namely `delete` and `delete_by`.
+
+`delete` would delete the first occurrence of a value in an iterator.
+`delete_by` acts like `delete`, but also lets the user specify their own comparator predicate.
+This is different from `filter` which removes all occurrences that match the predicate.
+
+# Motivation
+[motivation]: #motivation
+
+The motivation for this is simply to be able to filter out single copies of values, rather than all of them.
+
+I was in this exact situation myself, and had to make use of the destructive `Vec::remove` method.
+Having a non-destructive way of doing something similar that fits within the ethos of `Iterator` seems nicer.
+
+# Guide-level explanation
+[guide-level-explanation]: #guide-level-explanation
+
+The `delete` method removes the first occurrence of its input from the iterator.
```

Comment:
I think it would be helpful to add a method signature for `delete`. Perhaps:

```rust
fn delete<T>(self, x: T) where T: PartialEq<Self::Item> { ... }
```